### ISSUE TITLE
TypeScript type checking from t() function

### ISSUE THREAD

**Issue by MickaelAustoni** [i18next/next-i18next#2196]
## 💥 Regression Report

Since version 23.x.x  from 22.x.x, i have lost type checking autocomplete with `t` function from `useTranslation` react hook. 

## Last working version

Worked up to version: 22.X.X

Stopped working in version: 23.X.X

## To Reproduce

My `i18next.d.ts` :

```` TypeScript
import "i18next";
import en from "@/locales/en";

declare module "i18next" {
  interface CustomTypeOptions {
    returnNull: false;
    defaultNS: "translation";
    resources: {
      translation: typeof en;
    };
  }
}

````

Result :
![version23](https://github.com/i18next/next-i18next/assets/108873902/955b0039-1867-4257-8f90-3c18cc654d9b)

## Expected behavior

With the same config in 22.x.x :
![version22](https://github.com/i18next/next-i18next/assets/108873902/b0a70bb6-55ad-4c29-928a-fc4961b09913)

## Your Environment

- *runtime version*: node v18, chrome
- *i18next version*:  23.2.8
- *os*: Mac
- *IDE*: WebStorm

**Comment by adrai** [comment]
Seems to work here: https://github.com/i18next/next-i18next/tree/master/examples/simple

**Comment by MickaelAustoni** [comment]
Yes, the only difference I see in my case is that `vite` for my project and not `next`.

**Comment by adrai** [comment]
So why you are opening a github issue in this Next.js related repository if you use vite?

**Comment by MickaelAustoni** [comment]
Oh my bad sorry, it's a mistake ...

**Comment by adrai** [comment]
btw: maybe @rosskevin can help? https://github.com/i18next/react-i18next/issues/1635#issuecomment-1688914534
